lib/floki/selector/pseudo_class.ex

```diff
@@ -25,6 +25,17 @@ defmodule Floki.Selector.PseudoClass do
     false
   end
 
+  def match_contains?(tree, html_node, %__MODULE__{value: value}) do
+    res = Enum.filter(html_node.children_nodes_ids, fn(id) ->
```

Comment:
You could use `find` instead of `filter`, since we need to know if one or more `Text` nodes matches.

Another thing: it's more correct to check if the `value` is inside the `content` of the `Text` nodes.
The way it is, it will only match nodes that have only one word.

Example:

```elixir
html = """
<div>
  <p>roses are red, violets are blue</p>
</div>
"""

Floki.find(html, ":fl-contains(red)")
```

In this example, it won't match the "<p>" because the text node will include all the text (not only "red").

A way to match this would be something like:

```elixir
case Map.get(tree.nodes, id) do
  %Text{content: content} -> String.split(content, " ") |> Enum.any?(& &1 == value)
  _ -> false
end
```

Of course we have some ed-cases like the `,` (comma), and punctuation. We have to think in a way to avoid that.